lib/bozo/erubis_templating_coordinator.rb

```diff
@@ -120,9 +120,13 @@ def generate_file(template_path, block)
 
       content = @configuration.apply {|binding| template.result(binding)}
 
-      File.open(target_path, 'w+') {|f| f.write(content)}
+      if template_path.include? ".crt.template" || template_path.include? ".key.template"
```

Comment:
This class shouldn't be responsible for knowing about specific file names to change the mode. It's breaking behaviour, as someone else may have these filenames, and also it isn't sustainable when another template needs to specify whether the file is binary.

If a `binary` flag needs to differ for specific files then it should be defined by the `template_files`, which will change the `mode`.

```
prepare :file_templating do |t|
  t.template_files '{src,test}/**/*{.config,.json}.template'
  t.template_files '{src,test}/**/*{.key,.crt}.template', binrary: true
  t.exclude_files '{src,test}/**/obj/**/*{.config,.json,.key,.crt}.template'
end
```